### PR TITLE
Sidebar: Remove feature flag, enable by default, and deprecate "Show on click" option

### PR DESCRIPTION
**Pull request by simonhong**
Fix https://github.com/brave/brave-browser/issues/20876

With this change, sidebar will be available on all platform.
and sidebar show option is "Always" by default.

Security review - https://github.com/brave/security/issues/746

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used GitHub [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
`npm run test brave_unit_tests -- --filter=SidebarService*`

1. Launch browser and check sidebar is not shown
2. Check sidebar entry is visible from app menu and it has three options - show/hover/never

* Sidebar `On click` show option deprecation test
1. Launch prev brave version and set show option as On click and close.
2. Launch latest brave version and check show option is set as `On mouse over`

### CODE REVIEW

**Comment by simonhong** [comment]
Changed to Draft as some tests are failed.

File: browser/ui/brave_browser_window.cc [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/brave_browser_window.h"
+
```
Comment (bsclifton) [inline browser/ui/brave_browser_window.cc:7]: Can we have a comment here that these are provided for testing purposes? Maybe something like:
```
// Provide a base implementation (important for `TestBrowserWindow ` in tests)
// For real implementation, see `BraveBrowserView`
```

File: components/sidebar/sidebar_service.cc [hunk 2]
```diff
@@ -119,6 +116,17 @@ SidebarService::SidebarService(PrefService* prefs) : prefs_(prefs) {
 
 SidebarService::~SidebarService() = default;
 
+void SidebarService::MigrateSidearShowOptions() {
```
Comment (bsclifton) [inline components/sidebar/sidebar_service.cc:119]:
```suggestion
void SidebarService::MigrateSidebarShowOptions() {
```

File: components/sidebar/sidebar_service.h [hunk 3]
```diff
@@ -69,6 +69,7 @@ class SidebarService : public KeyedService {
   void UpdateSidebarItemsToPrefStore();
   std::vector<SidebarItem> GetDefaultSidebarItemsFromCurrentItems() const;
   void OnPreferenceChanged(const std::string& pref_name);
+  void MigrateSidearShowOptions();
```
Comment (bsclifton) [inline components/sidebar/sidebar_service.h:72]:
```suggestion
  void MigrateSidebarShowOptions();
```

File: components/sidebar/sidebar_service.cc [hunk 4]
```diff
@@ -97,19 +95,18 @@ std::vector<SidebarItem> GetDefaultSidebarItems() {
...
-      kSidebarShowOption, static_cast<int>(ShowSidebarOption::kShowAlways));
+      kSidebarShowOption, static_cast<int>(ShowSidebarOption::kShowNever));
   registry->RegisterIntegerPref(kSidebarItemAddedFeedbackBubbleShowCount, 0);
 }
 
 SidebarService::SidebarService(PrefService* prefs) : prefs_(prefs) {
   DCHECK(prefs_);
   LoadSidebarItems();
 
+  MigrateSidearShowOptions();
```
Comment (bsclifton) [inline components/sidebar/sidebar_service.cc:108]:
```suggestion
  MigrateSidebarShowOptions();
```

**Review by bsclifton — APPROVED** [review]
Left comments for a typo - but otherwise looks great 😄👍

File: components/sidebar/sidebar_service.cc [hunk 5]
```diff
@@ -97,19 +95,18 @@ std::vector<SidebarItem> GetDefaultSidebarItems() {
 
 // static
 void SidebarService::RegisterProfilePrefs(PrefRegistrySimple* registry) {
-  if (!base::FeatureList::IsEnabled(kSidebarFeature))
-    return;
-
   registry->RegisterListPref(kSidebarItems);
   registry->RegisterIntegerPref(
-      kSidebarShowOption, static_cast<int>(ShowSidebarOption::kShowAlways));
+      kSidebarShowOption, static_cast<int>(ShowSidebarOption::kShowNever));
```
Comment (bsclifton) [inline components/sidebar/sidebar_service.cc:100]: @rebron @petemill should we show sidebar by default?

**Review by bsclifton — CHANGES_REQUESTED** [review]
Left one comment for @rebron and @petemill - we have a chat on Slack. We may consider having Sidebar show by default 😄 

We can always change the decision later. If the user never modifies this setting, we can change the default again by setting a different value with `RegisterIntegerPref`. This will update the default properly for existing users as well as new users

File: components/sidebar/sidebar_service.cc (hunk 2 above)
Comment (simonhong) [reply components/sidebar/sidebar_service.cc]: Fixed

File: components/sidebar/sidebar_service.h (hunk 3 above)
Comment (simonhong) [reply components/sidebar/sidebar_service.h]: Fixed.

**Review by mkarolin — APPROVED** [review]
`chromium_src` changes LGTM

**Comment by simonhong** [comment]
Kindly ping to @iefremov :)

**Review by bsclifton — APPROVED** [review]

**Pull request merged**